### PR TITLE
Adding inferCNV/Ideogram visualization to React explore tab (SCP-3113)

### PR DESCRIPTION
**Pull request by bistline**
This updates restores the existing inferCNV/Ideogram visualization to the new React explore tab.  This only applies to studies that have previously run `inferCNV` in their study, and have synced outputs to add the Ideogram Annotations files to their study.  Note: it is possible to upload results to your study bucket and sync the results file by selecting the Ideogram Annotations file type and specifying the correct taxon/assembly.  This visualization can load independent of any clustering/annotations/expression data loaded into a study, and as such does not require any of the other file types to be present to enable this tab (which is now separate from the `genome` tab, which loads IGV only now).

Preview, using sample annotations taken from the [Demo: inferCNV oligodendroglioma](https://singlecell.broadinstitute.org/single_cell/study/SCP384/demo-infercnv-oligodendroglioma#study-visualize) study on production:

infercnv_ideogram

To test (once your user account has the `FeatureFlag` for `react_explore` enabled): 
1. Download the `ideogram_exp_means__Observations--Sample--group--cluster.json` file from [Demo: inferCNV oligodendroglioma](https://singlecell.broadinstitute.org/single_cell/study/SCP384/demo-infercnv-oligodendroglioma#study-visualize)
2. Place the file in a bucket for any study connected to your instance of SCP
3. Go to the "Sync Study" page for that study, and specify the following values for `ideogram_exp_means__Observations--Sample--group--cluster.json` and save:
    1. File type: `Ideogram Annotations`
    2. Species: `human`
    3. Assembly: `GRCh37` **!!Important or Ideogram won't render!!**
4. Click "View Study" at the top of the page
5. Click the "Genome (inferCNV)" tab and verify the output looks like the above screenshot

This PR satisfies SCP-3113

### CODE REVIEW

**Review by devonbush — APPROVED** [review]
This is great stuff--love the extra tests around the tab logic

**Comment by bistline** [comment]
Test failures unrelated (yet again).

File: app/javascript/components/visualization/InferCNVIdeogram.js [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    checkbox.addEventListener('click', () => {
+      updateTracks()
+    })
+  })
+}
+
+/**
+ * Note ideogram is unavailable for this numeric cluster
+ *
+ * Used in render_cluster.js.erb
```
Comment (eweitz) [inline app/javascript/components/visualization/InferCNVIdeogram.js:247]: Remove obsolete part of comment.

```suggestion
 * Note ideogram is unavailable for this numeric cluster
```

**Review by knapii-developments — APPROVED** [review]

**Review by eweitz — APPROVED** [review]
Looks good.  I noted a partly-obsolete comment.  

Although I suspect their causes are technically out of scope for this work, note how the gene search bar and view options panel have degraded UI for this use case in the React refactor, as shown below.  

Specifically:

* If no genes present in study, disable gene search text input.

* If metadata and clusters have not been uploaded, omit "Clustering" and "Annotation" dropdowns in View Options

### Before, without React
inferCNV_without_React_SCP_2021-03-17

### After, with React
inferCNV_with_React_SCP_2021-03-17

**Comment by bistline** [comment]
I agree that the state of the menus/controls can be refined more, but you are correct in that it should be tracked in a separate PR.

**Comment by bistline** [comment]
Test failure is unrelated and tracked in https://broadworkbench.atlassian.net/browse/SCP-3164 - local runs of `yarn ui-test` on this branch pass without issue:

```
Test Suites: 19 passed, 19 total
Tests:       57 passed, 57 total
Snapshots:   0 total
Time:        16.457 s
```

**Pull request merged**